psi4/driver/driver_findif.py

```diff
@@ -0,0 +1,795 @@
+            energies = np.hstack((energies, np.fliplr(energies)))
+        # Now determine all diagonal force constants for this irrep.
+        if data["num_pts"] == 3:
+            diag_fcs = energies[:, 0] + energies[:, 1]
+            diag_fcs -= 2 * ref_energy
+            diag_fcs /= (data["disp_size"]**2)
+        elif data["num_pts"] == 5:
+            diag_fcs = -energies[:, 0] + 16 * energies[:, 1] + 16 * energies[:, 2] - energies[:, 3]
+            diag_fcs -= 30 * ref_energy
+            diag_fcs /= (12 * data["disp_size"]**2)
+        H_irr = np.diag(diag_fcs)
+
+        # Step Two: Off-diagonals
+        num_unique_off_elts = n_salcs * (n_salcs - 1) // 2
+        offdiag_energies = irrep_energies[diag_disps:]
+        # If offdiagonal elements exist, put them into groups per salc pairs
+        # ...if offdiagonal elements DON'T exist, reshaping would raise an error.
+        if offdiag_energies:
+            offdiag_energies = np.reshape(offdiag_energies, (num_unique_off_elts, -1))
+        offdiag_row_index = 0
+
+        for i, salc in enumerate(salcs):
+            for j, salc2 in enumerate(salcs[:i]):
+                offdiag_row = offdiag_energies[offdiag_row_index]
+                if data["num_pts"] == 3:
+                    fc = (+offdiag_row[0] + offdiag_row[1] + 2 * ref_energy - energies[i][0] - energies[i][1] -
+                          energies[j][0] - energies[j][1]) / (2 * data["disp_size"]**2)
+                elif data["num_pts"] == 5:
+                    fc = (-offdiag_row[0] - offdiag_row[1] + 9 * offdiag_row[2] - offdiag_row[3] - offdiag_row[4] +
+                          9 * offdiag_row[5] - offdiag_row[6] - offdiag_row[7] + energies[i][0] - 7 * energies[i][1] -
+                          7 * energies[i][2] + energies[i][3] + energies[j][0] - 7 * energies[j][1] -
+                          7 * energies[j][2] + energies[j][3] + 12 * ref_energy) / (12 * data["disp_size"]**2)
+                H_irr[i, j] = fc
+                H_irr[j, i] = fc
+                offdiag_row_index += 1
+
+        B_pi.append(data["salc_list"].matrix_irrep(h))
+        H_pi.append(_process_hessian_symmetry_block(H_irr, B_pi[-1], massweighter, irrep_lbls[h], data["print_lvl"]))
+
+    # All blocks of the Hessian are now constructed!
+    return _process_hessian(H_pi, B_pi, massweighter, data["print_lvl"])
+
+
+def geoms_grad_from_energy(molecule):
+    """
+    Generate geometries for a gradient by finite difference of energies.
+    
+    Parameters
+    ----------
+    molecule : qcdb.molecule or psi4.core.Molecule
+        The molecule to compute the gradient of.
+
+    Returns
+    -------
+    disp_geoms : list of psi4.core.Matrix
+        A list of displaced geometries to compute energies at.
+
+    Notes
+    -----
+    Only symmetric displacements are necessary, so user specification of
+    symmetry is disabled.
+    """
+    return _geom_generator(molecule, -1, "1_0")
+
+
+def geoms_hess_from_grad(molecule, irrep):
```

Comment:
Up to you but `fd_2_1` is about as clear as the this line to me. Can we either go back to a shorthand or expand this out to `hessian_from_gradient_geometries` or something else very explicit.